test/e2e/framework/pod/utils.go

```diff
@@ -123,15 +133,25 @@ func GetLinuxLabel() *v1.SELinuxOptions {
 // DefaultNonRootUser is the default user ID used for running restricted (non-root) containers.
 const DefaultNonRootUser = 1000
 
+// DefaultNonRootUserName is the default username in Windows used for running restricted (non-root) containers
+const DefaultNonRootUserName = "ContainerUser"
+
 // GetRestrictedPodSecurityContext returns a restricted pod security context.
 // This includes setting RunAsUser for convenience, to pass the RunAsNonRoot check.
 // Tests that require a specific user ID should override this.
 func GetRestrictedPodSecurityContext() *v1.PodSecurityContext {
-	return &v1.PodSecurityContext{
+	psc := &v1.PodSecurityContext{
 		RunAsNonRoot:   pointer.BoolPtr(true),
-		RunAsUser:      pointer.Int64(DefaultNonRootUser),
+		RunAsUser:      GetDefaultNonRootUser(),
 		SeccompProfile: &v1.SeccompProfile{Type: v1.SeccompProfileTypeRuntimeDefault},
 	}
+
+	if NodeOSDistroIs("windows") {
```

Comment:
Since --node-os-distro is set as a e2e.test argument and we do schedule some linux containers during some Windows specific e2e tests.
This would add windows options with RunAsUserName set for those containers.
I don't think this will be an issue but did want to call that out.

Reply:
I looked at doing this but seems like it should really be done when `// TODO(#105919): Handle PodOS for windows pods.` is completed.  It made things a bit messier than it is now because there are a couple different cases too cover (`framework.NodeOSDistroIs` and handling `PodOS`since this needs to be addressed in `MixinRestrictedPodSecurity` as well.